File: ws/valencia/u1/ws08FromClassDiagramToCode/ws_08ChickenFarmSimulator/src/ec/edu/espe/chikenfarmsimulator/utils/FileManager.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.chikenfarmsimulator.utils;

import java.io.BufferedWriter;
import java.io.FileWriter;
import java.io.IOException;

/**
 *
 * @author USUARIO
 */
public class FileManager {
       public static void save(String data, String fileName, int type) {

        if (type == 0) { //TXT
            fileName = fileName + ".txt";
        }
        if (type == 1) { //CSV
            fileName = fileName + ".csv";
        }
        if (type == 2) { //JSON
            fileName = fileName + ".json";
        }

        try (FileWriter fileWriter = new FileWriter(fileName, true); BufferedWriter bufferedWriter = new BufferedWriter(fileWriter)) {

            bufferedWriter.write(data);
            bufferedWriter.newLine();

        } catch (IOException e) {
            System.err.println("Error pepito writing to the file: " + e.getMessage());
        }

    }
}

File: hw/ccedeno/u1/hw_10FileChicken/ChickenFarmJS/utils/FileManager.js
const fs = require('fs');

class FileManager {
    static save(data, fileName, type) {
        if (type === 0) { // TXT
            fileName = fileName + '.txt';
        } else if (type === 1) { // CSV
            fileName = fileName + '.csv';
        } else if (type === 2) { // JSON
            fileName = fileName + '.json';
        }

        fs.appendFile(fileName, data + '\n', (err) => {
            if (err) {
                console.error('Error writing to the file: ' + err.message);
            } else {
                console.log('Data successfully saved to ' + fileName);
            }
        });
    }
}
module.exports = FileManager;